modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -888,15 +888,21 @@ static void helpCallFunc(const iocshArgBuf *args)
                 "Type 'help <command>' to see the arguments of <command>.  eg. 'help db*'\n");
     }
     else {
+        bool printSeparator = false;
         for (int iarg = 1 ; iarg < argc ; iarg++) {
             for (pcmd = iocshCommandHead ; pcmd != NULL ; pcmd = pcmd->next) {
                 piocshFuncDef = pcmd->def.pFuncDef;
                 if (epicsStrGlobMatch(piocshFuncDef->name, argv[iarg]) != 0) {
-                    if(piocshFuncDef->usage) {
-                        fputs("\nUsage: ", epicsGetStdout());
+
+                    if (printSeparator) {
+                        fprintf(epicsGetStdout(), 
+                            ANSI_UNDERLINE("                                                            \n"));
+                    } else {
+                        printSeparator = true;
```

Comment:
This might be overly pedantic, but I find this logic a bit confusing. It ties the printing of a separator to whether or not we already have printed one. But the real reason we print a separator is whether or not we are printing multiple commands.

I would almost prefer that `printSeparator = true` is simply added unconditionally as the last command in the outer `for` loop. To me, this fits better with the reason that we add the separator, possibly with a separate name e.g.
```c
bool first_function = true;
for (arg in iarg) {
  for (pcmd in pcmds) {
    if (not first_function) {
      print_separator();
    }
  }
  first_function = false;
}
```
or something like that.

Reply:
Thank you for the feedback. I agree. Implemented in 8e49ba6ebb2388987deea9dbe1f12ea9e4a73b0b